service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/azure/DeleteAzureControlledResourceStep.java

```diff
@@ -1,5 +1,64 @@
 package bio.terra.workspace.service.resource.controlled.cloud.azure;
 
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.StepResult;
+import bio.terra.stairway.StepStatus;
+import bio.terra.workspace.common.exception.AzureManagementExceptionUtils;
 import bio.terra.workspace.service.resource.controlled.flight.delete.DeleteControlledResourceStep;
+import com.azure.core.management.exception.ManagementException;
+import java.util.Arrays;
+import java.util.List;
+import org.springframework.http.HttpStatus;
 
-public abstract class DeleteAzureControlledResourceStep implements DeleteControlledResourceStep {}
+public abstract class DeleteAzureControlledResourceStep implements DeleteControlledResourceStep {
+
+  @Override
+  public StepResult doStep(FlightContext context) throws InterruptedException {
+    try {
+      return deleteResource(context);
+    } catch (InterruptedException e) {
+      throw e;
+    } catch (Exception e) {
+      return handleResourceDeleteException(e, context);
+    }
+  }
+
+  protected abstract StepResult deleteResource(FlightContext context) throws InterruptedException;
```

Comment:
This should get a comment to document the intended usage for resource developers going forward. Maybe a note in the controlled_resources readme too explaining where this fits in when creating new resources?